cpp/order/RotationalAutocorrelation.h

```diff
@@ -0,0 +1,127 @@
+#include <tbb/tbb.h>
+
+#include "VectorMath.h"
+#include "HOOMDMath.h"
+#include "Index1D.h"
+
+/*! \file RotationalAutocorrelation.h
+    \brief Defines the RotationalAutocorrelation class, which computes the total
+    rotational autocorrelation for a system's orientations against its initial
+    orientations.
+*/
+
+namespace freud { namespace order {
+
+//! Convert a quaternion to complex coordinates.
+/*! \param q The quaternion to generate complex coordinates for.
+ *  \return A pair containing the coordinates (xi, zeta).
+ *
+ *  The set of quaternions are isomorphic to the special unitary group of
+ *  degree 2 SU(2), which forms a double cover of the 3D rotation group SO(3).
+ *  SU(2) is also diffeomorphic to the 3-sphere S3 (see
+ *  https://en.wikipedia.org/wiki/Special_unitary_group#Diffeomorphism_with_S3,
+ *  for example), meaning that we can represent quaternions in terms of two
+ *  complex numbers that map out hyperspherical coordinates in 3 dimensions.
+ *  This function generates that mapping.
+ */
+std::pair<std::complex<float>, std::complex<float> > quat_to_greek(const quat<float> &q);
+
+//! Compute a hyperspherical harmonic.
+/*! \param xi The first complex number coordinate.
+ *  \param zeta The second complex number coordinate.
+ *  \param l The azimuthal quantum number.
+ *  \param m1 The first magnetic quantum number.
+ *  \param m2 The second magnetic quantum number.
+ *  \return The value of the hyperspherical harmonic (l, m1, m2) at (xi, zeta).
+ *
+ *  The hyperspherical harmonic function is a generalization of spherical
+ *  harmonics from the 2-sphere to the 3-sphere. For details, see Harmonic
+ *  functions and matrix elements for hyperspherical quantum field models
+ *  (https://doi.org/10.1063/1.526210).
+ */
+std::complex<float> hypersphere_harmonic(const std::complex<float> xi, std::complex<float> zeta,
+                                          const int l, const int m1, const int m2);
+
+//! Compute the total rotational autocorrelation for a set of orientations.
+/*! The desired autocorrelation function is the rotational analog of the
+ *  dynamic structure factor, which provides information on the dynamcs of
+ *  systems of points. Calculating this quantity requires a generalization of
+ *  the Fourier transform to a different domain, namely the rotation group
+ *  SO(3). This computation can be performed using a hyperspherical coordinate
+ *  representation of the rotations. For details, see "Design rules for
+ *  engineering colloidal plastic crystals of hard polyhedra – phase behavior
+ *  and directional entropic forces" by Karas et al. (currently in preparation).
+ */
+class RotationalAutocorrelation
+    {
+    public:
+        //! Explicit default constructor for Cython.
+        RotationalAutocorrelation() {}
+
+        //! Constructor
+        /*! \param l The order of the spherical harmonic.
+         */
+        RotationalAutocorrelation(int l) : m_l(l), m_N(0), m_Ft(0) {}
```

Comment:
Should `l` be an `unsigned int`?

Reply:
Technically yes, but I think that it's easier to leave it signed so that we don't run into issues when we do things like compare it with signed `m1` and `m2` values. We could change it if you have a strong preference though.

Reply:
Ignore this - it's better as it is for the reason you mentioned. I missed the checks for "even and positive" in the Cython code during my first read-through.